runtime/src/blueoil.cpp

```diff
@@ -15,6 +17,130 @@
 
 
 namespace blueoil {
+
+int Tensor::shapeVolume() {
+    return this->shapeVolume(m_shape);
+}
+int Tensor::shapeVolume(std::vector<int> shape) {
+    return std::accumulate(shape.begin(), shape.end(),
+			   1, std::multiplies<int>());
+}
+
+Tensor::Tensor(std::vector<int> shape)
+    : m_shape(std::move(shape)),
+      m_data(std::vector<float>(this->shapeVolume(shape), 0)) {
+}
+Tensor::Tensor(std::vector<int> shape, std::vector<float> data)
+    : m_shape(std::move(shape)),
+      m_data(std::move(data)) {
+}
+Tensor::Tensor(std::vector<int> shape, float *arr)
+    : m_shape(std::move(shape)),
+      m_data(std::vector<float>(arr, arr + this->shapeVolume(shape))) {
+}
+Tensor::Tensor(const Tensor &tensor)
+    : m_shape(tensor.m_shape),
+      m_data(tensor.m_data) {
+}
+std::vector<int> Tensor::shape() {
+    return m_shape;
+}
+std::vector<int> Tensor::shape() const {
+    return m_shape;
+}
+float* Tensor::data() {
+    return &(m_data[0]);
```

Comment:
`m_data.data()` is more idiomatic way to do it.

Reply:
thank you.
https://github.com/blue-oil/blueoil/pull/9/commits/2a66a0f524f6f21f26edf654ba61b72f4d826dfc

Reply:
add >> blueoil::dataAt(std::vector<int> indices, bool clamp)
    del << blueoil::image::Tensor_at(blueoil::Tensor, int x, int y)
more generic function. dataAt(indices);
https://github.com/blue-oil/blueoil/pull/9/commits/790cfbabf07fbd2d9ca816df481f06865025dc7a

Reply:
By the way, I'm familiar with "AT" in OpenCV, so I used this name. It 's not good.
https://docs.opencv.org/3.4/d3/d63/classcv_1_1Mat.html#ac40425b84f60b39fd35e03814122661f